experimental/LieAlgebras/src/RootSystem.jl

```diff
@@ -164,6 +175,15 @@ function roots(R::RootSystem)
   return [[r for r in positive_roots(R)]; [-r for r in positive_roots(R)]]
 end
 
+function simple_root(R::RootSystem, i::Int)
+  @req 1 <= i <= rank(R) "Invalid index"
+  return positive_root(R, i)
+end
+
+function simple_roots(R::RootSystem)
+  return [positive_root(R, i) for i in 1:num_simple_roots(R)]
```

Comment:
```suggestion
  return positive_roots(R)[1:rank(R)]
```

Reply:
The current solution is more efficient than your suggestion. Your suggestion first creates a vector of all positive roots and then copies the first `rk` many to a new vector. As there are many more positive than simple roots, this is mostly useless work.

Reply:
No, the implementation is just an accessor with a type insertion. The suggestion instead prevents creating a new underlying array each time.

Reply:
Oh yeah, you are correct.